protocols/protocol/cloudstate/entity.proto

```diff
@@ -69,41 +109,70 @@ message SideEffect {
     // The payload of the command.
     google.protobuf.Any payload = 3;
 
-    // Whether this side effect should be performed synchronously, ie, before the reply is eventually
-    // sent, or not.
+    // Whether this side effect should be performed synchronously, ie, before the reply is eventually sent, or not.
     bool synchronous = 4;
 }
 
-// A command. For each command received, a reply must be sent with a matching command id.
+//Command represents TODO
+//
+//---
+//
+//`Specification`:
+//
+//  - An `Command` is considered to be valid iff:
+//    - `entity_id` TODO
+//    - `id` TODO
+//    - `name` TODO
+//    - `payload` TODO
+//    - `streamed` TODO
 message Command {
 
     // The ID of the entity.
     string entity_id = 1;
 
     // A command id.
+    // @exclude FIXME rename to command_id
     int64 id = 2;
 
-    // Command name
+    // Command name.
     string name = 3;
 
     // The command payload.
     google.protobuf.Any payload = 4;
 
-    // Whether the command is streamed or not
+    // Whether the command is streamed or not.
     bool streamed = 5;
 }
 
+//StreamCancelled TODO
+//
+//---
+//
+//`Specification`:
+//
+//  - A `StreamCancelled` is considered to be valid iff:
+//    - `entity_id` TODO
+//    - `id` TODO
 message StreamCancelled {
 
-    // The ID of the entity
+    // The ID of the entity.
     string entity_id = 1;
 
-    // The command id
+    // The command id.
+    // @exclude FIXME Rename to command_id
     int64 id = 2;
 }
 
-// A failure reply. If this is returned, it will be translated into a gRPC unknown
-// error with the corresponding description if supplied.
+//Failure represents a Command processing failure reply.
```

Comment:
What is the state of the User Function if such a Failure was sent? Is this a fatal situation (not recoverable)? Should the user function be "passivated"...?

Reply:
@marcellanz  Yes, that's a great question. We should outline that on the service, rather than the message, IMO. Wdyt?

Reply:
@viktorklang sure, that sounds very reasonable.